flutter/lib/device_info.dart

```diff
@@ -61,64 +63,56 @@ class DeviceInfo {
     if (machine != null) {
       modelName = deviceNames.getSingleNameFromModel(DeviceType.ios, machine);
     }
-    return DeviceInfo(
-      manufacturer: 'Apple',
-      androidInfo: null,
+    return EnvIos(
+      osVersion: Platform.operatingSystemVersion,
       modelCode: modelCode,
       modelName: modelName,
+      socName: getSocName(),
     );
   }
 
-  static Future<DeviceInfo> _makeAndroidInfo() async {
+  static Future<EnvAndroid> _makeAndroidInfo() async {
     final deviceInfo = await DeviceInfoPlugin().androidInfo;
     final deviceNames = DeviceMarketingNames();
 
     var shell = Shell();
-    final socModel = await shell.run('getprop ro.soc.model');
-    final socManufacturer = await shell.run('getprop ro.soc.manufacturer');
-
-    final androidInfo = EnvAndroidInfo(
-      propSocModel: socModel.outText,
-      propSocManufacturer: socManufacturer.outText,
-      buildBoard: deviceInfo.board,
-    );
+    final propSocModel = await shell.run('getprop ro.soc.model');
+    final propSocManufacturer = await shell.run('getprop ro.soc.manufacturer');
 
     final modelCode = deviceInfo.model;
     String? modelName;
     if (modelCode != null) {
       modelName =
           deviceNames.getSingleNameFromModel(DeviceType.android, modelCode);
     }
-    return DeviceInfo(
+    return EnvAndroid(
+      osVersion: Platform.operatingSystemVersion,
       // deviceInfo.manufacturer is usually a human-readable string with proper capitalisation
       // deviceInfo.brand seems to be machine-readable string because it tends to be all lower-case letters
       manufacturer: deviceInfo.manufacturer,
-      androidInfo: androidInfo,
       modelCode: modelCode,
       modelName: modelName,
+      boardCode: deviceInfo.board,
+      procCpuinfoSocName: getSocName(),
+      props: [
+        EnvAndroidProp(
+          type: 'soc_model',
+          name: 'ro.soc.model',
+          value: propSocModel.outText,
```

Comment:
`type` and `name` look redundant here? I would just define 2 fields `key` and `value` for `EnvAndroidProp`.
I guess you want to use the field `type` as a filter? Then it should be a `enum`, since it should have a small set of values.

Reply:
> I guess you want to use the field `type` as a filter? Then it should be a `enum`, since it should have a small set of values.

Yeah.

`name` field is useless here, As in: we will not show it, we will not use it for any logic.
But when we have working online result database we could use it to see which values are present on which devices.
Though, it should be noted that we are not saving the whole `getprop` output, we are only saving specified values, so will still need to manually determine list of values we are interested in, so the app saves them in result json.

Reply:
done